### DISCUSSION TITLE
[Valheim] monitor script rebooting server

### DISCUSSION THREAD

**Discussion by risteon1** [GameServerManagers/LinuxGSM#3325] [Q&A] [answered]
This only started after today's update. Looks like monitor script is failing on port 2457 query, which reboots server. You can see from the log file timestamps the frequency of it. I started playing today after 21:00UTC and experienced the disconnects from the reboots. Included below is output of one of the log files.

At this time I disabled the monitor script in crontab, and no issues seen while playing.
#*/5 * * * * /home/vhserver/vhserver monitor > /dev/null 2>&1

Any suggestions for a fix would be appreciated.

-rw-rw-r-- 1 vhserver vhserver    4811 Feb 26 09:30 vhserver-script-2021-02-26-09:30:01.log
-rw-rw-r-- 1 vhserver vhserver  783530 Mar  2 10:30 vhserver-script-2021-03-02-10:30:01.log
-rw-rw-r-- 1 vhserver vhserver    2954 Mar  2 10:41 vhserver-script-2021-03-02-10:40:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 10:51 vhserver-script-2021-03-02-10:50:02.log
-rw-rw-r-- 1 vhserver vhserver    3494 Mar  2 11:01 vhserver-script-2021-03-02-11:00:02.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 11:11 vhserver-script-2021-03-02-11:10:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 11:21 vhserver-script-2021-03-02-11:20:01.log
-rw-rw-r-- 1 vhserver vhserver    3494 Mar  2 11:31 vhserver-script-2021-03-02-11:30:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 11:41 vhserver-script-2021-03-02-11:40:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 11:51 vhserver-script-2021-03-02-11:50:01.log
-rw-rw-r-- 1 vhserver vhserver    3494 Mar  2 12:01 vhserver-script-2021-03-02-12:00:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 12:11 vhserver-script-2021-03-02-12:10:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 12:21 vhserver-script-2021-03-02-12:20:02.log
-rw-rw-r-- 1 vhserver vhserver    3494 Mar  2 12:31 vhserver-script-2021-03-02-12:30:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 12:42 vhserver-script-2021-03-02-12:40:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 12:51 vhserver-script-2021-03-02-12:50:02.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 13:01 vhserver-script-2021-03-02-13:00:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 13:11 vhserver-script-2021-03-02-13:10:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 13:21 vhserver-script-2021-03-02-13:20:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 13:31 vhserver-script-2021-03-02-13:30:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 13:41 vhserver-script-2021-03-02-13:40:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 13:51 vhserver-script-2021-03-02-13:50:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 14:02 vhserver-script-2021-03-02-14:00:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 14:11 vhserver-script-2021-03-02-14:10:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 14:21 vhserver-script-2021-03-02-14:20:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 14:31 vhserver-script-2021-03-02-14:30:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 14:42 vhserver-script-2021-03-02-14:40:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 14:51 vhserver-script-2021-03-02-14:50:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 15:01 vhserver-script-2021-03-02-15:00:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 15:12 vhserver-script-2021-03-02-15:10:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 15:22 vhserver-script-2021-03-02-15:20:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 15:31 vhserver-script-2021-03-02-15:30:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 15:41 vhserver-script-2021-03-02-15:40:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 15:51 vhserver-script-2021-03-02-15:50:02.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 16:01 vhserver-script-2021-03-02-16:00:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 16:11 vhserver-script-2021-03-02-16:10:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 16:21 vhserver-script-2021-03-02-16:20:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 16:31 vhserver-script-2021-03-02-16:30:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 16:41 vhserver-script-2021-03-02-16:40:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 16:51 vhserver-script-2021-03-02-16:50:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 17:01 vhserver-script-2021-03-02-17:00:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 17:11 vhserver-script-2021-03-02-17:10:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 17:21 vhserver-script-2021-03-02-17:20:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 17:31 vhserver-script-2021-03-02-17:30:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 17:41 vhserver-script-2021-03-02-17:40:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 17:51 vhserver-script-2021-03-02-17:50:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 18:01 vhserver-script-2021-03-02-18:00:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 18:11 vhserver-script-2021-03-02-18:10:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 18:21 vhserver-script-2021-03-02-18:20:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 18:31 vhserver-script-2021-03-02-18:30:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 18:41 vhserver-script-2021-03-02-18:40:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 18:51 vhserver-script-2021-03-02-18:50:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 19:01 vhserver-script-2021-03-02-19:00:02.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 19:12 vhserver-script-2021-03-02-19:10:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 19:21 vhserver-script-2021-03-02-19:20:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 19:31 vhserver-script-2021-03-02-19:30:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 19:41 vhserver-script-2021-03-02-19:40:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 19:51 vhserver-script-2021-03-02-19:50:02.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 20:01 vhserver-script-2021-03-02-20:00:02.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 20:11 vhserver-script-2021-03-02-20:10:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 20:21 vhserver-script-2021-03-02-20:20:02.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 20:31 vhserver-script-2021-03-02-20:30:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 20:42 vhserver-script-2021-03-02-20:40:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 20:51 vhserver-script-2021-03-02-20:50:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 21:01 vhserver-script-2021-03-02-21:00:01.log
-rw-rw-r-- 1 vhserver vhserver    3039 Mar  2 21:11 vhserver-script-2021-03-02-21:10:01.log
-rw-rw-r-- 1 vhserver vhserver    1836 Mar  2 21:18 vhserver-script-2021-03-02-21:18:43.log
-rw-rw-r-- 1 vhserver vhserver    3035 Mar  2 21:26 vhserver-script-2021-03-02-21:25:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 21:37 vhserver-script-2021-03-02-21:35:02.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 21:46 vhserver-script-2021-03-02-21:45:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 21:56 vhserver-script-2021-03-02-21:55:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 22:06 vhserver-script-2021-03-02-22:05:01.log
-rw-rw-r-- 1 vhserver vhserver    3326 Mar  2 22:16 vhserver-script-2021-03-02-22:15:01.log
-rw-rw-r-- 1 vhserver vhserver    3357 Mar  2 22:26 vhserver-script-2021-03-02-22:25:01.log
-rw-rw-r-- 1 vhserver vhserver    2889 Mar  2 22:36 vhserver-script-2021-03-02-22:35:01.log
-rw-rw-r-- 1 vhserver vhserver     207 Mar  2 22:36 vhserver-script.log
-rw-rw-r-- 1 vhserver vhserver     420 Mar  2 10:30 vhserver-steamcmd.log
vhserver@server1:~/log/script$ cat vhserver-script-2021-03-02-10:40:01.log

![vhserver-ss](https://user-images.githubusercontent.com/31777188/109726994-38d9f700-7b79-11eb-9fbc-29770199bf07.png)

**Comment by jp-powers** [comment, marked as answer]
this sounds like the same issue I'm seeing (https://github.com/GameServerManagers/LinuxGSM/discussions/3327), just with more specific info than I provided :-)

I'm disabling my monitor cronjob for now. There was a patch today that changed how dedicated servers work, namely "disconnecting" them from a Steam system, which I bet is related to the port. I'm also seeing that in the Steam client Servers list, where I added my server, it's not showing up during list refreshes now. Guessing if you set public to 0 it doesn't open the query port, which would otherwise be used to communicate to Steam that the server exists/is running, so the monitor check fails because it thinks it's needed?

**Reply by risteon1** [reply]
My server starts with the public 0 option, so that doesn't seem to make a difference.

**Reply by jp-powers** [reply]
Sorry, I was not clear on that. When I said, "Guessing if you set public to 0..." I meant the generic you, not you specifically.

To be more clear:

With this most recent update to the Valheim server, if a server is run with public 0, I expect that it is not running anything on what used to be the query port. I ran 'netstat -atunp | grep valheim' and got this:
(Not all processes could be identified, non-owned process info
 will not be shown, you would have to be root to see it all.)
tcp        0      0 <my server's local IP>:52913        <redacting one of my player's IPs>:27032    ESTABLISHED 141289/./valheim_se 
udp   119808      0 0.0.0.0:2457            0.0.0.0:*                           141289/./valheim_se 
udp6       0      0 :::2456                 :::*                                141289/./valheim_se

So it looks like it is opening the query port, but maybe it's not reporting the same data that it used to if public is 0. To my understanding, the query port is what the server would use so Steam can see it, know it exists, put it in the community servers lists in game, etc. Given that this new patch (patch notes here: https://store.steampowered.com/news/app/892970?emclan=103582791463217109&emgid=3058478454632856786) made changes to dedicated private servers (so like yours, riseteon1, and mine, both with public set to 0), and specifically that they no longer use Steam Datagram Relay, I expect that if the LGSM monitor script is looking for that in some degree it's getting a false negative, thus restarting the server.

I could definitely be way off the mark, of course. I just think it's interesting that with an app update that made what appears to be some fundamental changes to the way it does or doesn't use a particular third-party service, and then monitor breaking right after the update was applied on my system at least.

**Reply by risteon1** [reply]
Cool, I see what you mean. Yea, that makes sense. For now I'm running ok without monitor, so hopefully lgsm will figure out a fix so I can use it again.
